prebid-server/developers/add-a-module-go.md

```diff
@@ -0,0 +1,266 @@
+    }, hookstage.MutationUpdate, "bidrequest", "imp", "banner", "battr")
+	
+    return hookstage.HookResult[hookstage.BidderRequestPayload]{ChangeSet: changeSet}, nil
+}
+```
+
+2) To **reject** the bidder in the `BidderRequest` hook you would return:
+```
+func (m Module) HandleBidderRequestHook(
+	ctx context.Context,
+	invocationCtx hookstage.ModuleInvocationContext,
+	payload hookstage.BidderRequestPayload,
+) (hookstage.HookResult[hookstage.BidderRequestPayload], error) {
+	return hookstage.HookResult[hookstage.BidderRequestPayload]{Reject: true, NbrCode: 7}, nil
+}
+```
+
+Refer [here](https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/OpenRTB%20v3.0%20FINAL.md#list--no-bid-reason-codes-) for a list of available No Bid Response Codes.
+
+3) To supply [analytics tags](/prebid-server/developers/module-atags.html) in the `BidderRequest` hook you would return:
+```
+import (
+	"context"
+
+	"github.com/prebid/prebid-server/hooks/hookstage"
+	"github.com/prebid/prebid-server/hooks/hookanalytics"
+)
+
+func (m Module) HandleBidderRequestHook(
+	ctx context.Context,
+	invocationCtx hookstage.ModuleInvocationContext,
+	payload hookstage.BidderRequestPayload,
+) (hookstage.HookResult[hookstage.BidderRequestPayload], error) {
+    return hookstage.HookResult[hookstage.BidderRequestPayload]{
+        AnalyticsTags: hookanalytics.Analytics{
+            Activities: []hookanalytics.Activity{
+                {
+                    Name:   "enforce_blocking",
+                    Status: hookanalytics.ActivityStatusSuccess,
+                    Results: []hookanalytics.Result{
+                        {
+                            Status: hookanalytics.ResultStatusBlock,
+                            Values: map[string]interface{}{
+                                "attributes": []string{"bcat"},
+                                "bcat":       []string{"IAB-1"},
+                            },
+                            AppliedTo: hookanalytics.AppliedTo{Bidders: []string{"appnexus"}, ImpIds: []string{"imp_ID1"}},
+                        },
+                        {
+                            Status:    hookanalytics.ResultStatusAllow,
+                            AppliedTo: hookanalytics.AppliedTo{Bidders: []string{"appnexus"}, ImpIds: []string{"imp_ID2"}},
+                        },
+                    },
+                },
+            },
+        },
+    }, nil
+}
+```
+
+More test implementations for each hook can be found in unit-tests at [https://github.com/prebid/prebid-server/tree/master/modules/prebid/ortb2blocking](https://github.com/prebid/prebid-server/tree/master/modules/prebid/ortb2blocking) folder.
+
+### Configuration
+
+It's possible to define default module configuration which can be read by the module at PBS startup. Please see the [Configuration](https://docs.google.com/document/d/1CmamniQpwcI3p0_rHe2F17zV4sEhzpOdrqU7zuZVZ_I/edit#heading=h.mh3urph3k1mk) section of the technical specification.
+
```

Comment:
While configuring a module locally I had to read almost the entire document to find what exact config do I need to add. Maybe it makes sense to add simplest snippet of config here so this tutorial is self-sufficient and covers a basic scenario. The documentation will be excellent source to read if more complicated cases should be added. 

Here is a config I added (also for bidder request hook): 

```
"hooks": {
    "enabled": true,
    "modules":{
      "vendor1: {
        "module1": {
          "enabled": true
        }
      }
    },
    "host_execution_plan": {
      "endpoints": {
        "/openrtb2/auction":{
          "stages": {
            "bidder_request": {
              "groups": [
                {
                  "timeout": 10,
                  "hook_sequence": [
                    {
                      "module_code": "vendor1.module1",
                      "hook_impl_code": "code123"
                    }
                  ]
                }
              ]
            }
          }
        }
      }
    }
  }
```

Reply:
Sure, thanks for providing config 👍